opencti-platform/opencti-graphql/src/database/ai-llm.ts

```diff
@@ -22,6 +22,29 @@ const AI_MODEL = conf.get('ai:model');
 const AI_MAX_TOKENS = conf.get('ai:max_tokens');
 const AI_VERSION = conf.get('ai:version');
 
+/**
+ * Extracts the Azure instance name and deployment name from a URL of the form:
+ * https://<my-instance-name>.openai.azure.com/openai/deployments/<my-deployment-name>/
+ *
+ * @param {string} url - The full URL containing the Azure instance and deployment names.
+ * @returns {[string, string] | null} - Returns a tuple [azureInstance, azureDeployment] or null if not matched.
+ */
+export function extractAzureInstanceAndDeployment(
+  url: string,
+): [string, string] | null {
+  const regex = /^https:\/\/([^.]+)\.openai\.azure\.com\/openai\/deployments\/([^/]+)\/*/;
+  const match = url.match(regex);
+
+  if (!match) {
+    return null;
+  }
+
+  const [, azureInstance, azureDeployment] = match;
+  return [azureInstance, azureDeployment];
+}
+
+const [AI_AZURE_INSTANCE, AI_AZURE_DEPLOYMENT] = extractAzureInstanceAndDeployment(AI_ENDPOINT) ?? ['', ''];
```

Comment:
instance and deployment should be fetched from conf rather than from a parsed regex